config/src/main/java/com/typesafe/config/impl/ConfigNodeKey.java

```diff
@@ -0,0 +1,21 @@
+/**
+ *   Copyright (C) 2015 Typesafe Inc. <http://typesafe.com>
+ */
+package com.typesafe.config.impl;
+
+import java.util.Collection;
+
+final class ConfigNodeKey extends AbstractConfigNode {
```

Comment:
there's a convention throughout the lib, or at least in public API, that we use "key" to mean one element of a path; paths have to be parsed, keys are the raw strings in each Path element. ConfigNodeKey should probably be ConfigNodePath to be sure we don't get confused on this point.